src/orange/OrangeTrackView.hh

```diff
@@ -203,13 +204,23 @@ OrangeTrackView::operator=(const Initializer_t& init)
     local.temp_sense = this->make_temp_sense();
 
     // Initialize logical state
-    auto tracker = this->make_tracker(UniverseId{0});
-    auto tinit   = tracker.initialize(local);
-    // TODO: error correction/graceful failure if initialiation failured
-    CELER_ASSERT(tinit.volume && !tinit.surface);
+    UniverseId          uid{0};
+    VolumeId            global_vol_id{};
+    detail::UnitIndexer unit_indexer(params_.unit_indexer_data);
+
+    do
+    {
+        auto tracker = this->make_tracker(uid);
+        auto tinit   = tracker.initialize(local);
+        // TODO: error correction/graceful failure if initialiation failed
+        CELER_ASSERT(tinit.volume && !tinit.surface);
+        global_vol_id = unit_indexer.global_volume(uid, tinit.volume);
 
-    // Save local data
-    states_.vol[thread_] = tinit.volume;
+        OpaqueId<VolumeRecord> vrid(global_vol_id.unchecked_get());
+        uid = params_.volume_records[vrid].daughter;
```

Comment:
The antipattern of copying one OpaqueId value to another means we probably need to change `OrangeParamsData::volume_records` so that it's indexed by volume ID: `Collection<VolumeRecord, W, M, VolumeId>`. Then this will become:
```suggestion
        uid = params_.volume_records[global_vol_id].daughter;
```

Reply:
I was just doing what is already done in `univ/VolumeView.hh`, but I can fix that too. I see how a `Collection` of `VolumeRecord` can be created with `VolumeId` indexing. However, `SimpleUnitRecord` needs to store an `ItemRange` of this data. How I do go about templating `ItemRange` (which doesn't seem like it has a separate template argument for indexing) for this purpose?

Right now I am getting an error on:

``` 
unit.volumes = make_builder(&orange_data_->volume_records).insert_back(vol_records.begin(), vol_records.end());
```

because of this.

Reply:
Ooh, ok, that's a good point. Perhaps:
```c++
using VolumeRecordRange = Range<VolumeId>;
```
and have a `VolumeRecordRange` field?